plc4j/tools/connection-cache/src/main/java/org/apache/plc4x/java/utils/cache/LeasedPlcConnection.java

```diff
@@ -92,15 +102,139 @@ public PlcReadRequest.Builder readRequestBuilder() {
         if(connection == null) {
             throw new PlcRuntimeException("Error using leased connection after returning it to the cache.");
         }
-        return connection.readRequestBuilder();
+        final PlcReadRequest.Builder innerBuilder = connection.readRequestBuilder();
+        return new PlcReadRequest.Builder(){
+
+            @Override
+            public PlcReadRequest build() {
+                final PlcReadRequest innerPlcReadRequest = innerBuilder.build();
+                return new PlcReadRequest(){
+
+                    @Override
+                    public CompletableFuture<? extends PlcReadResponse> execute() {
+                        CompletableFuture<? extends PlcReadResponse> future = innerPlcReadRequest.execute();
+                        final CompletableFuture<PlcReadResponse> responseFuture = new CompletableFuture<>();
+                        future.handle((plcReadResponse, throwable) -> {
+                            if (plcReadResponse != null) {
+                                responseFuture.complete(plcReadResponse);
+                            } else {
+                                try {
+                                    destroy();
+                                } catch (Exception e) {
+                                }
+                                responseFuture.completeExceptionally(throwable);
```

Comment:
So do I understand this block correctly? If anything goes wrong with the request, it terminates the connection, right? Should we really be doing this? I mean ... it's easy to ask for an invalid tag ... that doesn't really make it necessary to reconnect? For example with my KNX gateway it only allows 3 connections in paralell and within a window of 2-3 minutes (when it closes inactive connections) ... not 100% sure we should be doing this.

Reply:
But I guess ... otherwise there's no way for the user to actually invalidate a connection ... I guess we should probably have different sets of Exceptions that we return. Some that indicate the connection is no longer usable and some that say: you did something wrong, please do it right next time (Like invalid addresses)

Reply:
Invalid tag is not throw exception. It is normal return result. This is unexcepted exception not caught by protocol. So this error is very serious network error or protocol data error

Reply:
Ok ... I think you are right with this. So please ignore my comment.